src/Agent.php

```diff
@@ -721,11 +722,11 @@ public function requestAgent($endpoint): Response
                 $response = $httpClient->request('GET', $endpoint, []);
                 self::$found_address = $address;
                 break;
-            } catch (\GuzzleHttp\Exception\RequestException $exception) {
-                // got an error response, we don't need to try other addresses
-                break;
-            } catch (\Throwable $exception) {
-                // many addresses will be incorrect
+            } catch (\GuzzleHttp\Exception\ClientException $exception) {
+                // If a 4XX error seems to be coming from an agent, we can stop trying. For every other exception, we can try other addresses
+                if (str_contains($exception->getResponse()->getHeaderLine('Server'), 'libwww-perl-daemon')) {
```

Comment:
Can you better check `if "Server" header contains "libwww-perl-daemon" OR "glpi-agent"` ?
Just to anticipate future versions of glpi-agent which won't use perl.